redis/subConn.go

```diff
@@ -164,24 +218,28 @@ func (c *subConn) resetConn() error {
 		}
 	}
 
-	prevConn := c.currConn
-	c.currConn = psc
-	if prevConn != nil {
-		prevConn.Close()
+	if s.currConn != nil {
+		s.cancelCurrMsgChan()
+		s.currConn.Close()
 	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	s.currConn = psc
+	s.msgChan = connReceiveChan(ctx, psc)
+	s.cancelCurrMsgChan = cancel
```

Comment:
⛏️⛏️⛏️

It is not immediately clear to me what `cancelCurrMsgChan` does mean.
What about `closeMsgChan`? I know we're not closing the channel, but it may make semantics clearer.

Reply:
Improved the name and overall handling of the channel [here](https://github.com/vtex/go-io/pull/43/commits/a0d5402d532babec9057867ab1c6236632157441)